cosmos/providers/dbt/core/operators/local.py

```diff
@@ -62,73 +52,24 @@ def run_command(
         cmd: list[str],
         env: dict[str, str],
     ) -> SubprocessResult:
-        # check project_dir
-        if self.project_dir is not None:
-            if not os.path.exists(self.project_dir):
-                raise AirflowException(
-                    f"Can not find the project_dir: {self.project_dir}"
-                )
-            if not os.path.isdir(self.project_dir):
-                raise AirflowException(
-                    f"The project_dir {self.project_dir} must be a directory"
-                )
-
-        # routing the dbt project to a tmp dir for r/w operations
-        target_dir = f"/tmp/dbt/{os.path.basename(self.project_dir)}"
-
-        changes = True  # set changes to true by default
-        if os.path.exists(target_dir):
-            # if the directory doesn't exist or if there are changes -- keep changes as true
-            comparison = dircmp(
-                self.project_dir, target_dir, ignore=["logs", "target", ".lock"]
-            )  # compares tmp and project dir
-            changes = has_differences(comparison)  # check for changes
-
-        # if there are changes between tmp and project_dir then copy them over
-        if changes:
-            logging.info(
-                f"Changes detected - copying {self.project_dir} to {target_dir}"
+        """
+        Copies the dbt project to a temporary directory and runs the command.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_project_dir = shutil.copytree(
+                self.project_dir, tmp_dir, dirs_exist_ok=True
```

Comment:
@jlaneve I think `dirs_exist_ok=True` can be removed since it should never be copying into a directory that already exists, and leaving it out also makes it compatible with Python 3.7 where that arg isn't implemented.

Reply:
@jbandoro good catch! updated